src/main/java/co/cask/hydrator/plugin/spark/dynamic/ScalaSparkCompute.java

```diff
@@ -134,16 +134,15 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
   public void initialize(SparkExecutionPluginContext context) throws Exception {
     String className = generateClassName(context.getStageName());
     interpreter = context.createSparkInterpreter();
-    File dir = config.getDependencies() == null ? null : Files.createTempDirectory("sparkprogram").toFile();
-    try {
-      if (config.getDependencies() != null) {
-        SparkCompilers.addDependencies(dir, interpreter, config.getDependencies());
-      }
+    if (config.getDependencies() != null) {
+      File tempDir = Files.createTempDirectory("sparkprogram").toFile();
+      SparkCompilers.addDependencies(tempDir, interpreter, config.getDependencies());
+      // TODO: Ideally we need to delete the temp dir when the pipeline exaction finished.
```

Comment:
Just created this JIRA to be added here: https://issues.cask.co/browse/CDAP-13001